compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/platform/PlatformWindowContext.desktop.kt

```diff
@@ -58,29 +59,54 @@ internal class PlatformWindowContext {
         _windowContainer = windowContainer
     }
 
-    fun setContainerSize(size: IntSize) {
-        if (_windowInfo.containerSize != size) {
-            _windowInfo.containerSize = size
-        }
+    fun setContainerSize(size: Size) {
+        _windowInfo.containerSize = IntSize(
+            width = size.width.roundToInt(),
+            height = size.height.roundToInt()
+        )
     }
 
-    /**
-     * Calculates the offset of the given [container] within the window.
-     * It uses [_windowContainer] as a reference for window coordinate space.
-     *
-     * @param container The container component whose offset needs to be calculated.
-     * @return The offset of the container within the window as an [IntOffset] object.
-     */
-    fun offsetInWindow(container: Component): IntOffset {
-        val scale = container.density.density
-        val pointInWindow = if (_windowContainer != null) {
-            SwingUtilities.convertPoint(container, Point(0, 0), _windowContainer)
+    fun calculatePositionInWindow(container: Component, localPosition: Offset): Offset {
+        val windowContainer = _windowContainer ?: return localPosition
+        return convertPoint(
+            position = localPosition,
+            fromView = container,
+            toView = windowContainer
+        )
+    }
+
+    fun calculateLocalPosition(container: Component, positionInWindow: Offset): Offset {
+        val windowContainer = _windowContainer ?: return positionInWindow
+        return convertPoint(
+            position = positionInWindow,
+            fromView = windowContainer,
+            toView = container
+        )
+    }
+
+    private fun convertPoint(position: Offset, fromView: Component, toView: Component): Offset {
+        return if (fromView != toView) {
+            val fromPoint = position.toPoint(fromView.density)
```

Comment:
No, we cannot do it in such way - it drops any possible transformations aside from simple translations

Reply:
Swing only do a translate transformation, if we look at the implementation of `SwingUtilities.convertPoint`

Reply:
I'd avoid relying on framework's implementation details

Reply:
But we have a real issue because of there implementation details - losing the precision.

Offset(4f, 4f) can become Offset(5f, 5f), but in reality it should be Offset(6f, 6f)

Reply:
Done